### DISCUSSION TITLE
Labels, HelperText and ErrorMessage for CheckboxGroup and RadioGroup

### DISCUSSION THREAD

**Discussion by Jonas-C** [chakra-ui/ark#2606] [Ideas]
(Originally posted on Discord)

Hi! I'm interested in creating a CheckboxGroupLabel to properly describe a CheckboxGroup. I'm assuming this'll be pretty straightforward. However, CheckboxGroup is an ark component, and not available in zag. Would adding more components make it eligible for inclusion in zag? 

Furthermore, I think both RadioGroup and CheckboxGroup could benefit from being able to mark the entire group invalid (and display an error message for the entire group). The same can be said about describing an entire group kind of like Field.HelperText already does. Wrapping an entire group within a Field.Root won't work, as the group items will also pick up any helper texts or errors, unless you wrap each item in more Field.Root. Should Field.Root support these scenarios, or should components like RadioGroup.HelperText be developed? Is this even needed?

**Comment by segunadebayo** [comment]
Good points @Jonas-C.

We intend to work on the Fieldset and Legend components as a follow-up to the Field PR. Checkbox and Radio groups can be considered a fieldset, and their labels are legends, making them semantically different from the Field/Label idea.

Perhaps a `Fieldset.Error` might be the component that renders the error for the fieldset. Just throwing ideas.

**Reply by Jonas-C** [reply]
I absolutely agree with this! Let me know if I can help out somehow:)

**Comment by segunadebayo** [comment]
Closing — CheckboxGroup and RadioGroup work with the `Fieldset` component which provides label, helper text, and error text.
